File: backend/entity/Payment/TransactionLog.go
package Payment

import (
	"time"
	"gorm.io/gorm"
)
type TransactionLog struct {
	gorm.Model
	TransactionDate	time.Time `json:"CourseDate"`

	StatusID uint `json:"StatusID"`
	PaymentStatus   PaymentStatus `gorm:"foreignKey:StatusID"`

	PaymentID uint `json:"PaymentID"`
	Payment   Payment `gorm:"foreignKey:PaymentID"`

	
}

File: frontend/src/Interface/TransactionLog.ts
import { PaymentInterface } from "./Payment";

export interface TransactionLogInterface {
    ID?: number;                    
    TransactionDate: Date;         
  
    StatusID: number;               
    PaymentStatus?: PaymentInterface; 
  
    PaymentID: number;              
    Payment?: PaymentInterface;     
  }
  